playbooks/roles/kubernetes/tasks/deploy-zowe-k8.yaml

```diff
@@ -0,0 +1,310 @@
+        service.kubernetes.io/ibm-load-balancer-cloud-provider-zone: "{{ k8s_service_annot[1] }}"
+        service.kubernetes.io/ibm-load-balancer-cloud-provider-vlan: "{{ k8s_service_annot[2] }}"
+    when: k8s_service_annot != none and k8s_service == "loadbalancer"
+  - name: Git checkout unmodified gateway-service-lb file
+    delegate_to: localhost
+    command: git checkout -- ../containers/kubernetes/samples/gateway-service-lb.yaml
+    ignore_errors: True
+  - name: Edit gateway loadbalancer file to update annotations
+    delegate_to: localhost
+    yedit:
+      src: ../containers/kubernetes/samples/gateway-service-lb.yaml
+      key: metadata.annotations 
+      value: 
+        service.kubernetes.io/ibm-load-balancer-cloud-provider-ip-type: "{{ k8s_service_annot[0] }}"
+        service.kubernetes.io/ibm-load-balancer-cloud-provider-zone: "{{ k8s_service_annot[1] }}"
+        service.kubernetes.io/ibm-load-balancer-cloud-provider-vlan: "{{ k8s_service_annot[2] }}"
+    when: k8s_service_annot != none and k8s_service == "loadbalancer"
+
+  - name: Deploy Gateway Service on Zowe K8s using loadbalancer 
+    delegate_to: localhost
+    kubernetes.core.k8s:
+      kubeconfig: "{{ kubeconfig }}" 
+      context: "{{ k8s_context }}"
+      state: present 
+      src: ../containers/kubernetes/samples/gateway-service-lb.yaml
+    when: k8s_service|lower == "loadbalancer"
+  - name: Deploy Discovery Service on Zowe K8s using loadbalancer
+    delegate_to: localhost
+    kubernetes.core.k8s:
+      kubeconfig: "{{ kubeconfig }}" 
+      context: "{{ k8s_context }}"
+      state: present 
+      src: ../containers/kubernetes/samples/discovery-service-lb.yaml
+    when: k8s_service|lower == "loadbalancer"
+  - name: Deploy Gateway Service on Zowe K8s uisng NodePort 
+    delegate_to: localhost
+    kubernetes.core.k8s:
+      kubeconfig: "{{ kubeconfig }}" 
+      context: "{{ k8s_context }}"
+      state: present 
+      src: ../containers/kubernetes/samples/gateway-service-np.yaml
+    when: k8s_service|lower == "nodeport"
+  - name: Deploy Discovery Service on Zowe K8s using NodePort
+    delegate_to: localhost
+    kubernetes.core.k8s:
+      kubeconfig: "{{ kubeconfig }}" 
+      context: "{{ k8s_context }}"
+      state: present 
+      src: ../containers/kubernetes/samples/discovery-service-np.yaml
+    when: k8s_service|lower == "nodeport" 
+
+  - name: Edit BareMetal Gateway Ingress file to update host info
+    delegate_to: localhost
+    yedit:
+      src: ../containers/kubernetes/samples/bare-metal/gateway-ingress.yaml
+      edits:
+      - key: spec.rules
+        value:
+          - host: "{{ k8s_gateway_domain }}"
+            http:
+             paths:
+               - path: /
+                 pathType: Prefix
+                 backend:
+                   service:
+                     name: gateway-service
+                     port:
+                       name: gateway-port
```

Comment:
Not sure if this works:

```yaml
yedit:
  edits:
  - key: spec.rules[0].host
    value: "{{ k8s_gateway_domain }}"
```

So we don't touch `spec.rules[0].http` section. The point is to avoid changing original definitions except it's really necessary to pick from host configuration.

Reply:
Sure.